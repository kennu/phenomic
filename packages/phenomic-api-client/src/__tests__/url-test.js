const url = require("../url")

describe("url", () => {

  it("should build URLs", () => {
    expect(url({ collection: "posts", id: "test" })).toMatchSnapshot()
    expect(url({ collection: "posts", by: "tag", value: "test" })).toMatchSnapshot()
    expect(url({ collection: "posts", by: "tag", value: "test", order: "asc" })).toMatchSnapshot()
    expect(url({ collection: "posts", by: "tag", value: "test", order: "asc", limit: 10 })).toMatchSnapshot()
    expect(url({ collection: "posts", by: "tag", value: "test", order: "asc", limit: 10, after: "BASE64" })).toMatchSnapshot()
    expect(url({ root: "local", collection: "posts", by: "tag", value: "test", order: "asc", limit: 10, after: "BASE64" })).toMatchSnapshot()
  })

})
