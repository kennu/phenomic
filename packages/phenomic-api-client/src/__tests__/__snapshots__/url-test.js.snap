// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`url should build URLs 1`] = `"//posts/item/test.json"`;

exports[`url should build URLs 2`] = `"//posts/by-tag/test/.json"`;

exports[`url should build URLs 3`] = `"//posts/by-tag/test/asc.json"`;

exports[`url should build URLs 4`] = `"//posts/by-tag/test/asc/limit-10.json"`;

exports[`url should build URLs 5`] = `"//posts/by-tag/test/asc/limit-10/after-BASE64.json"`;

exports[`url should build URLs 6`] = `"local/posts/by-tag/test/asc/limit-10/after-BASE64.json"`;
